### DISCUSSION TITLE
events  and browserClient.IsConnected

### DISCUSSION THREAD

**Discussion by PrestigeDevop** [Voltstro-Studios/UnityWebBrowser#313] [Q&A]
how to check out  on initialized and ready for the first time ? 

``` 
private void Awake()
{

    StartCoroutine(Example());

}
  IEnumerator Example()
  {
      yield return new WaitUntil(() => clientManager.browserClient.IsConnected==true);
      Debug.Log("DONE");
      print(Time.time);
      Ready();
  }

 public void LoadMySite()
 {
     string mywebsite = Application.streamingAssetsPath;
     string urlpath = mywebsite;
     string page = Path.Combine(urlpath, "index.html");
     if (File.Exists(page)) { Debug.Log("Found"); }
     else
     {
         Debug.Log(page);
         Debug.Log("error 404");
     }
     webBrowserClient.LoadHtml(mywebsite);
}


 private OnLoadStartDelegate Ready()
 {
     debug.print("Ready?");
     LoadMySite();
     throw new NotImplementedException();
 }


````

**Comment by PrestigeDevop** [comment]
okay , I found out that if you somehow call any browser method before it initialized or ready  the next line won't run in that script  /method  therefore if you see something like browser isn't connected or ready yet it's eventually it is like a break point  -return keyword  similar to js in my experience

**Reply by Voltstro** [reply]
A lot of methods will only work if UWB is connected and ready, otherwise they will throw either a UwbIsNotReadyException or a UwbIsNotConnectedException.

You can use [OnClientConnected](https://projects.voltstro.dev/UnityWebBrowser/latest/articles/user/events/#onclientconnected) event to invoke methods once UWB is both connected and ready.
